compile error 'CLOCK_BOOTTIME' undeclared when building in conda-forge

### Summary
Trying to build the current mbedtls for conda-forge, I get the following compile error:

( see https://dev.azure.com/conda-forge/84710dde-1620-425b-80d0-4cf5baca359d/_apis/build/builds/811489/logs/12 for full log)

work/library/platform_util.c:269:25: error: 'CLOCK_BOOTTIME' undeclared (first use in this function); did you mean 'CLOCK_REALTIME'?
2023-10-25T19:27:33.7496057Z   269 |     ret = clock_gettime(CLOCK_BOOTTIME, &tv);
2023-10-25T19:27:33.7496705Z       |                         ^~~~~~~~~~~~~~
2023-10-25T19:27:33.7497217Z       |                         CLOCK_REALTIME

### System information

Mbed TLS version (number or commit id): 
Operating system and version: 
Configuration (if not default, please attach `mbedtls_config.h`): 
Compiler and options (if you used a pre-built binary, please indicate how you obtained it): 
Additional environment information: 

### Expected behavior



### Actual behavior



### Steps to reproduce



### Additional information

--- a/ChangeLog.d/fix-linux-builds-in-conda-forge.txt
+++ b/ChangeLog.d/fix-linux-builds-in-conda-forge.txt
@@ -1,0 +1,2 @@
+Bugfix
+   * Fix build failure in conda-forge.  Fixes #8422.

--- a/library/platform_util.c
+++ b/library/platform_util.c
@@ -265,7 +265,7 @@ mbedtls_ms_time_t mbedtls_ms_time(void)
     struct timespec tv;
     mbedtls_ms_time_t current_ms;
 
-#if defined(__linux__)
+#if defined(__linux__) && defined(CLOCK_BOOTTIME)
     ret = clock_gettime(CLOCK_BOOTTIME, &tv);
 #else
     ret = clock_gettime(CLOCK_MONOTONIC, &tv);

